### DISCUSSION TITLE
Support for rustdesk in GLPI

### DISCUSSION THREAD

**Discussion by paspo** [rustdesk/rustdesk#1216] [Ideas]
Supremo (a competitor) has just been added to [GLPI](https://github.com/glpi-project/glpi).
How complex is to achieve similar support?

https://github.com/glpi-project/glpi-agent/issues/35

**Comment by rustdesk** [comment]
Need some one knowing GLPI well.

**Comment by paspo** [comment]
Adding support for just writing the ID is really simple, I've done it in 2 minutes straight on my glpi installation.
The missing parts are:

- support for direct link click (the software generates a link like `rustdesk:12345678` and the OS should open a new rustdesk connection with that ID). I suppose this is not supported at the moment.
- agent discovery of the rustdesk ID (primarily for windows machines, it's just a matter of retrieving the machine id)

None of these missing parts is required, but they are very important anyway.

**Reply by paspo** [reply]
[this](https://github.com/glpi-project/glpi-agent/pull/179/files) is how the agent manages the discovery for supremo.
As you can see, it's not hard.

**Reply by paspo** [reply]
I can take inspiration from the [Anydesk Inventory](https://github.com/glpi-project/glpi-agent/blob/develop/lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt/AnyDesk.pm) and use `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml` as path for the config file

**Reply by paspo** [reply]
[this](https://github.com/glpi-project/glpi-agent/compare/develop...paspo:glpi-agent:rustdesk) should go for the agent

**Reply by paspo** [reply]
and [this](https://github.com/glpi-project/glpi/compare/10.0/bugfixes...paspo:glpi:rustdesk) for the server part.

I didn't include the link to quickly open rustdesk and start a session because I think this is not supported at the moment.

**Comment by paspo** [comment]
- [x] [GLPI server support](https://github.com/glpi-project/glpi/pull/12406)
- [x] [GLPI agent support](https://github.com/glpi-project/glpi-agent/pull/193)
- [ ] [one-click session launch with the "rustdesk:" protocol - in rustdesk](https://github.com/rustdesk/rustdesk/pull/1241)
- [ ] one-click session launch with the "rustdesk:" protocol - in glpi

@rustdesk can we support last one (on Windows clients)?

it should be a matter of a "file association" that launches `rustdesk.exe` with a session code as a parameter

**Reply by paspo** [reply]
In https://github.com/rustdesk/rustdesk/issues/69 it says that `rustdesk --connect <id>` is working, and in fact it is!
It is not really documented...

**Reply by dbeniamine** [reply]
Hi,

I've done a script installing rustdesk on a linux computer setting a long random password and passing some additional content with a one click link in GLPI see [this message](https://github.com/glpi-project/glpi-agent/discussions/749#discussioncomment-10614476)

The only issue is that additional-contents appends the content to the one created by the agent, thus on GLPI I see two links, one with only the id and one with `"id?password=...."`

**Comment by billgatos** [comment]
Just to say, that rustdesk is recognized as a software, but the id isn't saved as a remote management id

**Comment by paspo** [comment]
I think this PR https://github.com/glpi-project/glpi-agent/pull/193 is still in the agent's development branch.

The affected file is not present in 1.4 ([see here](https://github.com/glpi-project/glpi-agent/tree/1.4/lib/GLPI/Agent/Task/Inventory/Generic/Remote_Mgmt)).

You have to wait or compile your agent from sources or wait for the next release.

**Reply by C-Duv** [reply]
IDs are (now?) encrypted, how do you decode them in GLPI Agent?

**Comment by billgatos** [comment]
hi all,

The following recepie works:
GLPI 10.0.5 and agent 1.5 nightly gets the rustdesk id to GLPI asset.
